Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/TokenizingTextBox.cs

```diff
@@ -381,6 +374,21 @@ public void AddTokenItem(object data, bool atEnd = false)
             _ = AddTokenAsync(data, atEnd);
         }
 
+        /// <summary>
+        /// Clears the whole collection, will raise the <see cref="TokenItemRemoving"/> event asynchronously for each item.
+        /// </summary>
+        /// <returns>async task</returns>
+        public async Task Clear()
+        {
+            while (_innerItemsSource.Count > 1)
+            {
+                var container = ContainerFromItem(_innerItemsSource[0]) as TokenizingTextBoxItem;
+                await RemoveToken(container);
```

Comment:
We have to be careful here, as the developer can cancel the removal in the TokenItemRemoving event... which means this would loop forever.

Should we have an additional flag to have it not raise the removing event in the Clear case? (it can still fire the Removed event)

Reply:
if the item removal is cancelled then the Clear() operation should halt as well.
I'll look at how to detect the cancel and do the right thing.

Reply:
fixed. Added a UT for this as well.
I found that when running the UX UT containers for items in the list are not available. I made some small mods to the interface and implementation so I can still test without containers.